hydromt/metadata_resolvers/convention_resolver.py

```diff
@@ -0,0 +1,109 @@
+"""MetaDataResolver using HydroMT naming conventions."""
+from itertools import product
+from string import Formatter
+from typing import TYPE_CHECKING, Any
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+
+from hydromt.nodata import NoDataStrategy
+from hydromt.typing import Bbox, TimeRange
+
+from .metadata_resolver import MetaDataResolver
+
+if TYPE_CHECKING:
+    from hydromt.data_sources.data_source import DataSource
+
+
+class ConventionResolver(MetaDataResolver):
+    """MetaDataResolver using HydroMT naming conventions."""
+
+    _uri_placeholders = {"year", "month", "variable", "zoom_level"}
+
+    def _expand_uri_placeholders(
+        self, uri: str, time_tuple: tuple[str, str] | None, variables: set[str] | None
+    ) -> tuple[str, list[str]]:
+        """Expand known placeholders in the URI."""
+        keys: list[str] = []
+
+        if "{" in uri:
+            uri_expanded = ""
+            for literal_text, key, fmt, _ in Formatter().parse(uri):
+                uri_expanded += literal_text
+                if key is None:
+                    continue
+                key_str = "{" + f"{key}:{fmt}" + "}" if fmt else "{" + key + "}"
+                # remove unused fields
+                if key in ["year", "month"] and time_tuple is None:
+                    uri_expanded += "*"
```

Comment:
are these globs or more place holders for later?

Reply:
globs

Reply:
whill this cause issues in for example API endpoints?

Reply:
I wonder if we can use these placeholders in case of querying APIs. I would then point the URI to the API and have the reset of the settings as driver kwargs.